qutebrowser/browser/shared.py

```diff
@@ -306,9 +306,13 @@ def netrc_authentication(url, authenticator):
         # altogether.
         return False
     user, password = None, None
+    try:
+        host = url.host()
+    except AttributeError:
+        host = url
```

Comment:
This kind of thing definitely isn't the right way to go about this :wink:

What about just making all callers pass a host (rather than a URL), and then do `url.host()` where `netrc_authentication` is called instead?